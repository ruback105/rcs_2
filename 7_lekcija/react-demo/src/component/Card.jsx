import Avatar from "./Avatar";

function Card({ initials, fullName, profession, onDelete }) {
  return (
    <div className="border-2 rounded-md border-black p-2 flex justify-between">
      <div className="flex">
        <Avatar>{initials}</Avatar>
        <div className="flex flex-col ml-1 space-y-1">
          <h3 className="text-lg font-semibold">{fullName}</h3>
          <p className="text-sm">{profession}</p>
        </div>
      </div>

      <button
        onClick={() => {
          onDelete();
        }}
        className="border rounded-md py-2 px-4 border-red-700 text-red-700"
      >
        Delete
      </button>
    </div>
  );
}

export default Card;
