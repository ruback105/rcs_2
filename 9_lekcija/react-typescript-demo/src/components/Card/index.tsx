import { FC } from "react";

type CardType = {
  title: string;
  description: string;
};

const Card: FC<CardType> = ({ title, description }) => (
  <div>
    <h2>{title}</h2>
    <p>{description}</p>
    <p data-testid="additional-content">Additional content</p>
    {[1, 2, 3].map((item) => (
      <p key={item}>{item}</p>
    ))}
  </div>
);

export default Card;
