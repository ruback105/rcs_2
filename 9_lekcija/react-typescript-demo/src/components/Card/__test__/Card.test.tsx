import { render, screen } from "@testing-library/react";
import Card from "../";
import React from "react";
import renderer from "react-test-renderer";

const title = "Some title";
const description = "Some description";

describe("Card component", () => {
  it("should render Card component", () => {
    render(<Card title={title} description={description} />);

    expect(screen.getByText(title).textContent).toEqual(title);
    expect(screen.getByText(description).textContent).toEqual(description);
  });

  it("should render additional content", () => {
    const testId = "additional-content";
    const additionalContent = "Additional content";

    render(<Card title="" description="" />);

    expect(screen.getByTestId(testId).textContent).toEqual(additionalContent);
  });

  it("should match snapshot", () => {
    const component = renderer.create(
      <Card title={title} description={description} />
    );

    const tree = component.toJSON();

    expect(tree).toMatchSnapshot();
  });
});
