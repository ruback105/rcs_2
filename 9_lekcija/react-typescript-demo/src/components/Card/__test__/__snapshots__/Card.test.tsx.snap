// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card component should match snapshot 1`] = `
<div>
  <h2>
    Some title
  </h2>
  <p>
    Some description
  </p>
  <p
    data-testid="additional-content"
  >
    Additional content
  </p>
</div>
`;
